scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1182,8 +1182,24 @@ def cmake_assemble_args(extra_args, handler, extra_conf_files, extra_overlay_con
         return args_expanded
 
     def cmake(self, filter_stages=[]):
+        args = []
+        for va in self.testsuite.extra_args.copy():
+            cond_args = va.split(":")
+            if cond_args[0] == "arch" and len(cond_args) == 3:
+                if self.instance.platform.arch == cond_args[1]:
+                    args.append(cond_args[2])
+            elif cond_args[0] == "platform" and len(cond_args) == 3:
+                if self.instance.platform.name == cond_args[1]:
+                    args.append(cond_args[2])
+            elif cond_args[0] == "simulation" and len(cond_args) == 3:
+                if self.instance.platform.simulation == cond_args[1]:
+                    args.append(cond_args[2])
+            else:
```

Comment:
maybe could be worth to raise ConfigurationError, or at least log an error message, if len(cond_args) > 1, to avoid passing wrong extra_args to cmake.

Reply:
good suggestion. let me add